src/impl/android/external_commit_helper.cpp

```diff
@@ -106,7 +106,18 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
         }
         // the fifo already existing isn't an error
         if (ret == -1 && err != EEXIST) {
-            throw std::system_error(err, std::system_category());
+            // Workaround for a mkfifo bug on Blackberry devices:
+            // When the fifo already exists, mkfifo fails with error ENOSYS which is not correct.
```

Comment:
Should we be doing this extra test with `stat` only if the error is `ENOSYS`, rather than for all errors?